lib/xandra/cluster/control_connection.ex

```diff
@@ -445,11 +445,17 @@ defmodule Xandra.Cluster.ControlConnection do
               source: :xandra
             })
 
-            send(data.cluster, {:host_added, host})
-            Map.put(acc, peername, %{host: host, status: :up})
+            {existing_acc, Map.put(discovered_acc, peername, %{host: host, status: :up})}
         end
       end)
 
+    discovered_hosts =
+      Map.values(discovered_peers) |> Enum.map(fn %{host: %Host{} = host} -> host end)
```

Comment:
If we don't care about the `:status`, why don't we store the discovered peers as a list of `%Host{}` directly